### PR TITLE
Applied enhancement proposals

### PR DESCRIPTION
**Pull request by clementh59**
Fixed:
- https://github.com/evrythng/epcis2.js/issues/7
- https://github.com/evrythng/epcis2.js/issues/9
- https://github.com/evrythng/epcis2.js/issues/8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -264,6 +266,20 @@ o.clearEPCList();
 const object = o.toObject(); //{ isA: 'ObjectEvent'} -> the epcList isn't sent anymore
 ```
 
+### Building the URN
+
+The SDK allows developers to build easily gs1 URI.
```
Comment (C-D-Lewis) [inline README.md:271]:
```suggestion
The SDK allows developers to build easily a GS1 URI. 
```

File: README.md [hunk 2]
```diff
@@ -264,6 +266,20 @@ o.clearEPCList();
 const object = o.toObject(); //{ isA: 'ObjectEvent'} -> the epcList isn't sent anymore
 ```
 
+### Building the URN
+
+The SDK allows developers to build easily gs1 URI. 
+
+For example, you can easily build a SGTIN URI:``
```
Comment (C-D-Lewis) [inline README.md:273]:
```suggestion
For example, you can easily build a SGTIN URI:
```

File: package.json [hunk 3]
```diff
@@ -114,7 +114,7 @@
     "@babel/runtime": "^7.13.17",
     "ajv": "^6.12.6",
     "cross-fetch": "^3.1.4",
-    "crypto-js": "^4.0.0",
+    "crypto-js": "^3.3.0",
```
Comment (C-D-Lewis) [inline package.json:117]: Why did this go back a major version?

File: src/entity/events/Event.js [hunk 4]
```diff
@@ -107,8 +107,22 @@ export default class Event extends Entity {
...
     if (settings.eventTimeZoneOffset !== undefined) {
       this.setEventTimeZoneOffset(settings.eventTimeZoneOffset);
+    } else if (this.eventTimeZoneOffset === undefined) {
+      const date = new Date();
+      const timeZoneOffset = getTimeZoneOffsetFromStringOrNumber(date.getTimezoneOffset() / 60);
+      this.setEventTimeZoneOffset(timeZoneOffset);
+    }
+
+    // If the event time isn't defined, set default values
+    if (this.getEventTime() === undefined) {
```
Comment (C-D-Lewis) [inline src/entity/events/Event.js:120]:
```suggestion
    if (!this.getEventTime()) {
```

Comment (C-D-Lewis) [reply src/entity/events/Event.js:120]: Can also be applied above

File: src/entity/events/Event.js [hunk 5]
```diff
@@ -532,6 +548,9 @@ export default class Event extends Entity {
     if ((typeof readPoint) === 'string') { // the param is the id of the readPoint
       this.readPoint = new ReadPoint().setId(readPoint);
       return this;
+    } if (!(readPoint instanceof ReadPoint)) {
```
Comment (C-D-Lewis) [inline src/entity/events/Event.js:551]:
```suggestion
    } 
    if (!(readPoint instanceof ReadPoint)) {
```

File: src/entity/events/Event.js [hunk 6]
```diff
@@ -555,6 +574,9 @@ export default class Event extends Entity {
     if ((typeof bizLocation) === 'string') { // the param is the id of the bizLocation
       this.bizLocation = new BizLocation().setId(bizLocation);
       return this;
+    } if (!(bizLocation instanceof BizLocation)) {
```
Comment (C-D-Lewis) [inline src/entity/events/Event.js:577]:
```suggestion
    }
    if (!(bizLocation instanceof BizLocation)) {
```

File: test/events/objectEvent.spec.js [hunk 7]
```diff
@@ -19,6 +19,7 @@ import DestinationElement from '../../src/entity/model/DestinationElement';
 import SensorElement from '../../src/entity/model/sensor/SensorElement';
 import Ilmd from '../../src/entity/model/Ilmd';
 import { exampleObjectEvent } from '../data/eventExample';
+import { getTimeZoneOffsetFromStringOrNumber } from '../../src/utils/utils';
```
Comment (C-D-Lewis) [inline test/events/objectEvent.spec.js:22]: `FromStringOrNumber` is evident from the inputs, so could shorten the name of this function.

File: test/events/objectEvent.spec.js [hunk 8]
```diff
@@ -39,13 +40,15 @@ describe('unit tests for the ObjectEvent class', () => {
     it('should not use eventTimeZoneOffset from settings if it is not overridden', async () => {
       setup({});
       const o = new ObjectEvent();
-      expect(o.eventTimeZoneOffset).to.be.equal(undefined);
+      expect(o.eventTimeZoneOffset).to.be.equal(
+        getTimeZoneOffsetFromStringOrNumber((new Date()).getTimezoneOffset() / 60),
```
Comment (C-D-Lewis) [inline test/events/objectEvent.spec.js:44]: This might only pass in the `-1` timezone. So once we leave BST, it could break if run in London.

**Review by C-D-Lewis — CHANGES_REQUESTED** [review]

File: test/events/objectEvent.spec.js (hunk 8 above)
Comment (clementh59) [reply test/events/objectEvent.spec.js:44]: Are you sure?
o.eventTimeZoneOffset is set to the local timezone and `(new Date()).getTimezoneOffset() / 60` retrieve the value (in hours) of the timezoneOffset

**Review by C-D-Lewis — APPROVED** [review]

**Pull request merged**